math/src/field/traits.rs

```diff
@@ -50,11 +50,21 @@ pub trait HasFieldOperations: Debug {
     // TODO: This are not exactly operations they are constructors
     // maybe they should be in another trait "HasFieldConstructors", and this trait should
     // require that one.
-    
+
     /// Returns the element `x * 1` where 1 is the multiplicative neutral element.
     fn from_u64(x: u64) -> Self::BaseType;
 
     /// Takes as input an element of BaseType and returns the internal representation
     /// of that element in the field.
     fn from_base_type(x: Self::BaseType) -> Self::BaseType;
 }
+
+pub trait IsLinearField: HasFieldOperations {
+    fn shr(a: &Self::BaseType, n: usize) -> Self::BaseType;
+
+    fn shl(a: &Self::BaseType, n: usize) -> Self::BaseType;
+
+    fn and(a: &Self::BaseType, mask: usize) -> usize;
+
+    fn bit_size() -> usize;
+}
```

Comment:
> Having bit operations is not a property of `LinearFields`, so this may be confusing. And the dependency with `HasFieldOperations` is not necessary. Just implementing the std ops and asking for them in the function is enough. If you want to aggregate them in a type is good too.
> 
> The only thing missing would be `bit_size`. We may need a trait for this, or add it in another place. In any scenario, it would be better to rename it to something like `bits` or `num_bits`
> 
> The mask of the and the function should be a `BaseType` too, since the number may be bigger than an usize.

@MauroToscano, got into a rabbit hole here. 
First, some quick answers to your points:
- I changed `LinearFields` to not extend `HasFieldOperations`.
- For shift operations, I removed them from `LinearFields` and asked for those bounds in the `impl` blocks. Looks better.
- Renamed `bit_size` to `num_bits`, and also renamed `LinearFields` to `SizedField` (maybe that's a better name?).

But, I have one last thing in `SizedField` (previously called `LinearField`) which I'd want to remove: the `and` operation.

As you suggested, it would be nice to:
- Make this implementable with `std::ops` as I did with the shift operations, which worked nicely.
- Maybe make the `mask` be `Self::BaseType`, etc.

I tried to do this for some time, but I think I'm stuck trying to achieve that. The current state of this PR still works, but it needs this last extra work that I can't seem be able to do. I have a [separate branch (single commit)](https://github.com/jsign/lambdaworks/commit/4cbb49184da66779b1e5e570e789383c8c3b6c68) with the furthest I could go implementing this, but I think is a dead end.

First, I realized that trying to make `mask` be `Self::BaseType` as you suggest it shouldn't be necessary, and it complicates things a bit:
- "Not necessary" (IMHO) since `window_sizes` in Pippenger are a couple of bits long, and AFAIK big windows sizes aren't used (but I could be wrong). 
- "Complicate things a bit", since we need the result of the `&` to be `usize` since the result of that operation is used in `buckets : Vec<...>` as `buckets[and_result]` so it should be `usize`.

As mentioned before, I tried hard to be able to accommodate these constraints, but I've honestly failed to do it, probably because of my lacking Rust-type system skills!

If you can, try to pull [separate branch (single commit)](https://github.com/jsign/lambdaworks/commit/4cbb49184da66779b1e5e570e789383c8c3b6c68) (is a separate branch), and see that while I got everything to compile (i.e: `cargo build`), while doing `cargo test` I get to a fundamental problem of Rust `u64` not implementing `BitAnd<F>` for `F` that isn't strictly `u64` (i.e: `usize`).

Maybe I'm running in circles here and you can imagine an easy solution, but I think after some time of trying I'm stuck. 
Can I get a bit of help here? I'd love to see how this can be done, since this last `and` in `SizedField` is the last remaining nit to make this PR ready!